src/mococrw/ecies.h

```diff
@@ -0,0 +1,332 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+#include "symmetric_crypto.h"
+#include "mac.h"
+#include "kdf.h"
+#include "asymmetric_crypto_ctx.h"
+
+namespace mococrw {
+
+class ECIESEncryptionCtxBuilder;
+
+/**
+ * @brief ECIESEncryptionCtx
+ *
+ * This class supports encryption of plain texts using ECIES
+ *
+ */
+class ECIESEncryptionCtx {
+public:
+    /**
+     * @brief Destructor
+     */
+    ~ECIESEncryptionCtx();
+
+    /**
+     * Encrypt a chunk of data.
+     *
+     * This function stores the data in the internal buffer.
+     * The data in the internal buffer is encrypted, when the finish method is invoked.
+     *
+     * @param message chunk of data to encrypt.
+     * @throws MoCOCrWException if this function is invoked after finish was called.
+     */
+    void update(const std::vector<uint8_t>& message);
+
+    /**
+     * @brief Finalize the encryption.
+     *
+     * This includes:
+     * 1. Creating the ephemeral key used for encryption (use getEphemeralKey() for getting it)
+     * 2. Encrypt the plaintext
+     * 3. Calculate the tag/mac (use getMac() for getting it)
+     *
+     * After invoking this function the ephemeral key and the authentication key can obtained using the getter
```

Comment:
```suggestion
     * After invoking this function, the ephemeral key and the authentication key can obtained using the getter
```
I'd add a comma here :-D